zilliqa/src/consensus.rs

```diff
@@ -750,6 +751,14 @@ impl Consensus {
                     .map(|tx_nonce| tx_nonce >= account_nonce)
                     .unwrap_or(true)
             })
+            .take_while(|txn| {
+                if let Some(g) = gas_left.checked_sub(txn.tx.gas_limit()) {
+                    gas_left = g;
+                    true
+                } else {
+                    false
+                }
+            })
```

Comment:
I'm assuming (haha) that there is an additional check which stops txns which have tx.gas_limit() > BLOCK_GAS_LIMIT  entering the pool (because they will now never leave it), and that best_transaction() orders by gas descending (otherwise txns can starve - and worse, I can starve them with enough gas - though perhaps you're OK with that?)

Reply:
There isn't, but there is an issue to add it :) #1003  However just to clarify, the current behaviour (in this PR) is that transactions with `gas_limit > BLOCK_GAS_LIMIT` will immediately be flushed from the pool because `checked_sub` will always return false.


Yes, `best_transaction()` always orders by gas descending.